app/controllers/assignments_controller.rb

```diff
@@ -570,7 +570,7 @@ def starter_file_group_file_data(starter_file_group)
       if (starter_file_group.path + file).directory?
         { key: "#{file}/" }
       else
-        { key: file, size: 1,
+        { key: file, size: 1, submitted_date: l(Time.current),
```

Comment:
@JunruL It seems that you've found the right place where file data is being sent, but you're sending the wrong value. The "Submitted at" time should be based on the property of the file that's been actually saved on disk, not just the current time.